### ISSUE TITLE
Does not add "data-qa-*" attribute to all React Components

### ISSUE THREAD

**Issue by besrabasant** [suprraz/babel-plugin-react-element-info#3]
It works fine. But does not add **data-qa** attributes to all _React Components_ . 
I think the issue is with the _Components_ that are loaded imported dynamically.

**Comment by ujh** [comment]
It doesn't work for me either. I've yet to figure out when that happens though. My issue is that in many cases you end up with stuff like `<div data-qa-node="div" data-qa-file="correct-file-name" >...</div>`. My best guess right now is that this happens for components that are just arrow functions and not class components?

**Comment by besrabasant** [comment]
For now, I am just reverting back to plain old **id** attribute

**Comment by camlegleiter** [comment]
I'm seeing a similar issue, but only for the `data-qa-file` attribute to include the name of the file (minus extension), though this was after an upgrade to various Babel modules/plugins.

Before:
```
"babel-core": "^6.21.0",
"babel-plugin-transform-decorators-legacy": "^1.3.4",
"babel-plugin-transform-runtime": "^6.15.0",
"babel-preset-es2015": "^6.18.0",
"babel-preset-es2016": "^6.22.0",
"babel-preset-es2017": "^6.22.0",
"babel-preset-react": "^6.16.0",
"babel-preset-stage-0": "^6.16.0",
"babel-register": "^6.9.0",
"react-hot-loader": "3.0.0-beta.7",
```

After:
```
"@babel/plugin-proposal-class-properties": "^7.0.0-beta.54",
"@babel/plugin-proposal-decorators": "^7.0.0-beta.54",
"@babel/plugin-proposal-export-default-from": "^7.0.0-beta.54",
"@babel/plugin-transform-runtime": "^7.0.0-beta.54",
"@babel/preset-env": "^7.0.0-beta.54",
"@babel/preset-react": "^7.0.0-beta.54",
"@babel/register": "^7.0.0-beta.54",
"@babel/runtime": "^7.0.0-beta.54",
"react-hot-loader": "^4.3.4",
```

I've identified a fix for making sure `data-qa-file` gets added since the attributes are only added when parsing JSX and certain path/state contents are present (though I have little knowledge of what "path" and "state" look like/get set/etc., nor how these properties are set and passed along). I'll push up a PR to address this. I'm not sure if it helps with @besrabasant or @ujh.

**Comment by Pjstaab** [comment]
@camlegleiter Did you ever push this PR up? We are running into the same problem.

**Comment by camlegleiter** [comment]
Created a potential fix: #6
